### ISSUE TITLE
Grep search result list: the file path is too long, the matched text are truncated

### ISSUE THREAD

**Issue by kangshugang** [vifm/vifm#483]
First, I would thank you guys very much for developing such an amazing tool that is super cool and super useful. I like it so much!

But I am experiencing a small issue. In the grep search result, the search results are always truncated from the right. This is not big deal, when the file paths are short. But sometimes, when the file paths are very long, then the result display is not good: the long long file path are displayed in full, but the matched text are almost all truncated. And there is no horizontal scroll capability. In such a situation, it would be much better to truncate the file paths from the left first (only keep the file name visible, but truncate the directories), thus make room available to display the matched text, which is much more useful than the directories.

Actually, I have fixed this issue, and here is the code change for your reference. If you agree with me. Maybe, you can integrate this change in the future versions.

Here is the code change (added in menu.c):
```
 489   if(utf8_strsw(item_tail) > (size_t)(width - 2))
 490   {
 491     char *p2 = strchr(item_tail, ':');
 492     char *p1 = p2;
 493     if (p2)
 494     {
 495       while (p1 > item_tail)
 496       {
 497         if (*p1 == '/') break;
 498         p1--;
 499       }
 500       int offset = (p1 - item_tail);
 501       if (offset > 0)
 502       {
 503         int len = utf8_strsw(item_tail);
 504         int target_len = width - 2 > len - offset + 3 ? width - 2 : len - offset + 3;
 505         char *ellipsed = left_ellipsis(item_tail, target_len, curr_stats.ellipsis);
 506         free(item_tail);
 507         item_tail = ellipsed;
 508       }
 509     }
 510   }
```
Here is a screen capture of the location of the code:
![Screen Shot 2019-10-09 at 9 39 18 PM](https://user-images.githubusercontent.com/19381920/66540205-f1644e00-eadf-11e9-9550-83f874f82726.png)

Thank you!

**Comment by xaizek** [comment]
Thanks for the code, an option to control alignment might be worth considering. Different kinds of menus might need different defaults.

> And there is no horizontal scroll capability.

There is actually. See the [docs](https://vifm.info/vimdoc.shtml#vifm-m_zh) for <kbd>zh</kbd>, <kbd>zl</kbd>, <kbd>zH</kbd> and <kbd>zL</kbd> keys in menus.

**Comment by kangshugang** [comment]
Thanks

**Comment by PRESFIL** [comment]
Is this issue resolved? Looks like yes.

**Comment by xaizek** [comment]
The patch wasn't applied. If I remember correctly it breaks functionality.

**Comment by PRESFIL** [comment]
Yes, but the problem that they tried to solve with this patch (it was not very correct to solve it) has already been solved by the presence of `zh` `zl` `zH` `zL`. It was suggested by mistake. So I think this question is closed.
